src/runtime/types.cpp

```diff
@@ -727,9 +729,17 @@ extern "C" int PySlice_GetIndices(PySliceObject* r, Py_ssize_t length, Py_ssize_
     Py_FatalError("unimplemented");
 }
 
-extern "C" int PySlice_GetIndicesEx(PySliceObject* r, Py_ssize_t length, Py_ssize_t* start, Py_ssize_t* stop,
+extern "C" int PySlice_GetIndicesEx(PySliceObject* _r, Py_ssize_t length, Py_ssize_t* start, Py_ssize_t* stop,
                                     Py_ssize_t* step, Py_ssize_t* slicelength) noexcept {
-    Py_FatalError("unimplemented");
+    try {
+        BoxedSlice* r = (BoxedSlice*)_r;
+        assert(r->cls == slice_cls);
+        parseSlice(r, length, start, stop, step, slicelength);
```

Comment:
would you mind verifying that parseSlice has the exact same behavior as this function?  I think it's supposed to, but without a comment saying that it is I feel wary of assuming that it is.